strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/logging/LoggingManagementController.java

```diff
@@ -0,0 +1,229 @@
+    {
+        try
+        {
+            Path logsBaseDir = Paths.get(propertiesBooter.getLogsDirectory());
+            Path requestedLogPath = Paths.get(logsBaseDir.toString(), path.orElse(""));
+
+            logger.debug(String.format("Requested directory listing of path: [%s] resolved to [%s]", path,
+                                       requestedLogPath));
+
+            if (!Files.exists(requestedLogPath))
+            {
+                return getNotFoundResponseEntity("Requested path does not exist!", acceptHeader);
+            }
+            if (!Files.isDirectory(requestedLogPath))
+            {
+                return getBadRequestResponseEntity("Requested path is not a directory!", acceptHeader);
+            }
+
+            DirectoryListing directoryListing = getDirectoryListingService().fromPath(logsBaseDir, requestedLogPath);
+
+            if (acceptHeader != null && acceptHeader.contains(MediaType.APPLICATION_JSON_VALUE))
+            {
+                return ResponseEntity.ok(objectMapper.writer().writeValueAsString(directoryListing));
+            }
+
+            String currentUrl = StringUtils.chomp(request.getRequestURI(), "/");
+            String downloadUrl = currentUrl.replaceFirst("/browse", "/download");
+            boolean showBack = path.isPresent() && !StringUtils.isBlank(path.get());
+
+            model.addAttribute("showBack", showBack);
+            model.addAttribute("currentUrl", currentUrl);
+            model.addAttribute("downloadBaseUrl", downloadUrl);
+            model.addAttribute("directories", directoryListing.getDirectories());
+            model.addAttribute("files", directoryListing.getFiles());
+
+            return new ModelAndView("directoryListing", model);
+        }
+        catch (Exception e)
+        {
+            String message = "Attempt to browse logs failed. Check server logs for more information.";
+            return getExceptionResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR, message, e, acceptHeader);
+        }
+    }
+
+    @ApiOperation(value = "Used to stream logging file.")
+    @GetMapping(value = "/stream", produces = { org.carlspring.strongbox.net.MediaType.TEXT_EVENT_STREAM_UTF8_VALUE,
+                                                org.carlspring.strongbox.net.MediaType.TEXT_PLAIN_UTF8_VALUE })
+    public SseEmitter logFileStream()
+            throws IOException
+    {
+        final SseEmitter sseEmitter = new SseEmitter(sseTimeoutMillis);
+        Resource logFileResource = getLogFileResource();
+        if (logFileResource == null)
+        {
+            sseEmitter.completeWithError(
+                    new IllegalStateException("Missing 'logging.file' or 'logging.path' properties"));
+            return sseEmitter;
+        }
+
+        ForkJoinPool forkJoinPool = ForkJoinPool.commonPool();
+        forkJoinPool.execute(
+                Tailer.create(logFileResource.getFile(),
```

Comment:
we probably can have `Tailer` bean with `@Scope(proxyMode = ScopedProxyMode.INTERFACES, value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)` instead of using `applicationContext.getBean(..)` here.

Reply:
Actually `applicationContext.getBean(..)` relates to `SseEmitterAwareTailerListenerAdapter`. I'll check if field injection will work instead of  `applicationContext.getBean(..)`

Reply:
finally I used the `Factory method` solution for `SseEmitterAwareTailerListenerAdapter`, thanks for suggestion :)

Reply:
> Actually applicationContext.getBean(..) relates to SseEmitterAwareTailerListenerAdapter. I'll check if field injection will work instead of applicationContext.getBean(..)

yes, I understand

the proposal was to make `Tailer` a bean like below:
```
        @Bean
        @Scope(proxyMode = ScopedProxyMode.TARGET_CLASS, value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
        public Tailer logFileResourceTailer(Resource logFileResource) {
            //create the `SseEmitterAwareTailerListenerAdapter` and `Tailer` instances here
           return tailer;
        }

```

this way it will be a proxy which creates new `Tailer` instance every time when `Tailer.run()` method invoked.

But anyway this is just an possible improvement, factory method probably suits as well.